pkg/npm/npm.go

```diff
@@ -0,0 +1,63 @@
+package npm
+
+import (
+	"bytes"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"io"
+	"strings"
+)
+
+type NpmRegistryOptions struct {
+	DefaultNpmRegistry string
+	SapNpmRegistry     string
+}
+
+type runner interface {
+	SetEnv(e []string)
+	Stdout(out io.Writer)
+}
+
+type execRunner interface {
+	runner
+	RunExecutable(e string, p ...string) error
+}
+
+func SetNpmRegistries(options *NpmRegistryOptions, execRunner execRunner) error {
+	const sapRegistry = "@sap:registry"
+	const npmRegistry = "registry"
+	configurableRegistries := []string{npmRegistry, sapRegistry}
+	for _, registry := range configurableRegistries {
+		var buffer bytes.Buffer
+		execRunner.Stdout(&buffer)
+		err := execRunner.RunExecutable("npm", "config", "get", registry)
+		execRunner.Stdout(log.Writer())
+		if err != nil {
+			return err
+		}
+		preConfiguredRegistry := buffer.String()
+
+		log.Entry().Info("Discovered pre-configured npm registry " + preConfiguredRegistry)
```

Comment:
The log output is perhaps a bit misleading, as apparently this can start with "undefined".  I would also include in the output, which of the registries was tested.